x-pack/plugins/secops/server/lib/network/query_top_n_flow.dsl.ts

```diff
@@ -260,3 +271,13 @@ const getOppositeField = (networkTopNFlowType: NetworkTopNFlowType) => {
     return NetworkTopNFlowType.client;
   }
 };
+
+const getQueryOrder = (networkTopNFlowSortField: NetworkTopNFlowSortField) => {
+  if (networkTopNFlowSortField.field === NetworkTopNFlowFields.bytes) {
+    return { bytes: networkTopNFlowSortField.direction };
+  } else if (networkTopNFlowSortField.field === NetworkTopNFlowFields.packets) {
+    return { packets: networkTopNFlowSortField.direction };
+  } else if (networkTopNFlowSortField.field === NetworkTopNFlowFields.ipCount) {
+    return { ip_count: networkTopNFlowSortField.direction };
+  }
+};
```

Comment:
Looking at this block of code and `getOppositeField` above it as well as similar patterns, the issue I see with them is that the implicit return value includes `undefined` since the if statement is not `exhaustive`. The problem there is any code which calls this will have to protect against `undefined` and usually that ends up by turning type checking using an exclamation point.

There are a couple of ways of dealing with this. My suggestion is to use the `never` type and move this over to a switch statement as that is more readable and really what this if else if is doing like so:

```ts
const assertUnreachable = (x: never): never => {
  throw new Error(`Unknown Field in switch statement ${x}`);
};

type QueryOrder = { bytes: Direction } | { packets: Direction } | { ip_count: Direction };

const getQueryOrder = (networkTopNFlowSortField: NetworkTopNFlowSortField): QueryOrder => {
  switch (networkTopNFlowSortField.field) {
    case NetworkTopNFlowFields.bytes:
      return { bytes: networkTopNFlowSortField.direction };
    case NetworkTopNFlowFields.packets:
      return { packets: networkTopNFlowSortField.direction };
    case NetworkTopNFlowFields.ipCount:
      return { ip_count: networkTopNFlowSortField.direction };
  }
  assertUnreachable(networkTopNFlowSortField.field);
};
```

I would do the same thing with the other transformations of fields. It would be better to blow up during runtime with a human readable stack trace than with a `undefined` error.

Reply:
maybe I am missing something here but the field is from an enum so how it can become undefined.

Reply:
Talk to @FrankHassanabad and @andrew-goldstein , I do understand now what it is meaning